src/yarp.c

```diff
@@ -8019,7 +8019,11 @@ parse_statements(yp_parser_t *parser, yp_context_t context) {
             while (accept_any(parser, 2, YP_TOKEN_NEWLINE, YP_TOKEN_SEMICOLON));
             if (context_terminator(context, &parser->current)) break;
         } else {
-            expect(parser, YP_TOKEN_NEWLINE, "Expected a newline or semicolon after statement.");
+            if (parser->heredoc_end) {
```

Comment:
I don't think we want to handle this here. I think it's probably likely that we're handling the escape somewhere and not flushing the `heredoc_end` as we should be.

Maybe take a look where we are currently handling `\` in `YP_LEX_DEFAULT`? I'd bet we aren't checking for a heredoc.

Reply:
@HParker if you don't mind removing this section, I think it's good.